### PR TITLE
Fixes #30571 - Title has already been taken during first time reposit…

### PR DESCRIPTION
**Pull request by sjha4**
To Test:
1. Create an OS under Hosts> Operating System with Description - **CentOS 7**
2. Create and sync repo - http://mirror.centos.org/centos/7/os/x86_64/
On this branch you shouldn't see the error.

### CODE REVIEW

**Comment by sjha4** [comment]
Took a long time to get this on pulp3 but the operating system does get created with pulp3 centos7 sync. 
Error on master:
Screenshot from 2020-09-10 12-07-17
Should be fixed with the change as with Pulp2

**Review by jjeffers — APPROVED** [review]
Tested this and was able to create the OS and sync the indicated repository - no issues!

File: app/models/katello/concerns/redhat_extensions.rb [hunk 1]
```diff
@@ -8,8 +8,10 @@ def find_or_create_operating_system(repo)
           os_name = construct_name(repo.distribution_family)
           major, minor = repo.distribution_version.split('.')
           minor ||= '' # treat minor versions as empty string to not confuse with nil
-
-          create_os = lambda { ::Redhat.where(:name => os_name, :major => major, :minor => minor).first_or_create! }
+          os = ::Redhat.where(:name => os_name, :major => major, :minor => minor).try(:first)
+          return os if os
+          description = "#{os_name}-#{repo.distribution_version}"
+          create_os = lambda { ::Redhat.where(:name => os_name, :major => major, :minor => minor, :description => description).first_or_create! }
```
Comment (parthaa) [inline app/models/katello/concerns/redhat_extensions.rb:14]:
```suggestion
          create_os = lambda { ::Redhat.where(:name => os_name, :major => major, :minor => minor, :description => description).create! }
```

Comment (sjha4) [reply app/models/katello/concerns/redhat_extensions.rb]: Updated..

**Comment by parthaa** [comment]
ack

**Pull request merged**